File: Java/src/main/java/Solution601_700/Solution657.java
package Solution601_700;

/**
 * @author windows
 */
public class Solution657 {
    public boolean judgeCircle(String moves) {
        int[] UDLR = new int[4];
        for (int i = 0; i < moves.length(); i++) {
            if (moves.charAt(i) == 'U'){
                UDLR[0]++;
            }else if (moves.charAt(i) == 'D'){
                UDLR[1]++;
            }else if (moves.charAt(i) == 'L') {
                UDLR[2]++;
            }else{
                UDLR[3]++;
            }
        }
        return UDLR[0] == UDLR[1] && UDLR[2] == UDLR[3];
    }


    public static void main(String[] args) {
        Solution657 solution657 = new Solution657();
    }
}

File: C++/src/Solution601_700/Solution657.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution657.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool judgeCircle(string moves) {
  int horizon = 0;
  int vertical = 0;
  for (const auto &item : moves){
    switch (item) {
      case 'L' : horizon--;
        break;
      case 'R' : horizon++;
        break;
      case 'U' : vertical++;
        break;
      case 'D' : vertical--;
        break;
    }
  }
  return horizon == 0 && vertical == 0;
}